pkg/prometheus/operator.go

```diff
@@ -19,17 +19,47 @@ import (
 	"reflect"
 	"regexp"
 	"strings"
+	"time"
 
 	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1alpha1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1alpha1"
+	monitoringclient "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	"github.com/prometheus-operator/prometheus-operator/pkg/informers"
+	"github.com/prometheus-operator/prometheus-operator/pkg/k8sutil"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
 
 	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	ResyncPeriod = 5 * time.Minute
 )
 
 var prometheusKeyInShardStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)-shard-[1-9][0-9]*$")
 var prometheusKeyInStatefulSet = regexp.MustCompile("^(.+)/prometheus-(.+)$")
 
+type CommonConfig struct {
+	ClusterConfig *rest.Config
+	KClient       *kubernetes.Clientset
+	MClient       *monitoringclient.Clientset
+
+	PromInfs   *informers.ForResource
+	PromAgInfs *informers.ForResource
+	SmonInfs   *informers.ForResource
+	PmonInfs   *informers.ForResource
+	ProbeInfs  *informers.ForResource
+	CmapInfs   *informers.ForResource
+	SecrInfs   *informers.ForResource
+	SsetInfs   *informers.ForResource
```

Comment:
I think that it makes sense to have a struct regrouping informers for servicemonitors, podmonitors, probes, confimaps, statefulsets and secrets .
On the other side, prometheus and prometheusagents informers need to live in their respective controllers.